Mathlib/Topology/MetricSpace/HolderNorm.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2024 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import Mathlib.Topology.MetricSpace.Holder
+
+/-!
+# Hölder norm
+
+This file defines the Hölder (semi-)norm for Hölder functions alongside some basic properties.
+
+## Main definitions
+
+* `eHolderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f`.
+* `nnHolderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0` of a function `f`.
+* `MemHolder r f`: `f` is `r`-Hölder continuous.
+
+## Main results
+
+* `eHolderNorm_eq_zero_iff`: the Hölder norm of a function is zero if and only if it is constant.
+* `MemHolder.holderWith`: The Hölder norm of a Hölder function `f` is a Hölder constant of `f`.
+
+## Tags
+
+Hölder norm, Hoeder norm
+
+-/
+
+variable {X Y Z : Type*}
+
+open Filter Set
+
+open NNReal ENNReal Topology
+
+section PseudoEMetricSpace
+
+variable [PseudoEMetricSpace X] [PseudoEMetricSpace Y] {r : ℝ≥0} {f : X → Y}
+
+/-- The `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f` is the least non-negative real
+number `C` for which `f` is `r`-Hölder continuous with constant `C`. This is `∞` if no such
+non-negative real exists. -/
+noncomputable
+def eHolderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0∞ := ⨅ (C) (_ : HolderWith C r f), C
```

Comment:
https://en.wikipedia.org/wiki/Germanic_umlaut#Substitution says Hölder should be transcribed to Hoelder in ASCII
```suggestion
def eHoelderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0∞ := ⨅ (C) (_ : HolderWith C r f), C
```

Reply:
I will make a separate PR changing all the names (including `HolderWith` and all the file names) after this one

Reply:
I have opened #16870. I would strongly suggest the name now instead of in a later PR, so that we avoid needing tons of deprecations

Reply:
Please make #16870 depend on this PR since you will need to rename some items from this PR

Reply:
Ah, it will probably go through before yours